File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var min=prices[0], max=0;
    for(let i=1;i<prices.length;i++){
         if(min>prices[i]){
            min=prices[i];
         }
         max=Math.max(max,prices[i]-min)
    }
    return max;
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min=prices[0];
        int maxPf=0;
        for(int i=0;i<prices.length;i++){
            maxPf=Math.max(maxPf,prices[i]-min);
            min=Math.min(prices[i],min);
        }
       return maxPf;
    }
}